use streaming/iterators, swap eyes, fix audio, add option to use depth video file

silphendio opened the pull request:
I changed the program to work with iterators, which means everything is processed frame by frame and my PC can convert a whole movie without running out of memory.

While I was doing that, I also set the correct FPS for the output video,
fixed the audio (I think moviepy doesn't want to write over a file it's also reading from),
and swapped the views so that the left view is on the left.

I got rid of the depth data cache too. Instead I added an option to read the depth map from a video file.

Sorry about putting so much stuff into the PR. I started with the streaming/iterators and changed some other stuff along the way.
I noticed too late that the `fixing_app` branch is newer. It should be possible to port these changes over.

silphendio committed:
use streaming to save memory, set correct fps, make audio work

silphendio committed:
switch eyes

silphendio committed:
fix broken iterator after saving depth video

silphendio committed:
remove obsolete progress bar

silphendio committed:
eliminate depth pickle file
also use more iterators, generators, and tees

silphendio committed:
add option to read depth fom video file, light refactoring

silphendio committed:
fix save_depth_only
added some if-statements
switched things around so tees are only created when they are used
(tees allocate memory when they are out of sync)

otanadzetsotne commented:
@silphendio thanks for your improvements. I checked your changes and it looks fine but can I ask you how you tested your solution?
Is the audio and codecs copied correctly, and is the final video supported by VR headsets, which are the target platform for this project? Unfortunately, I don't have time to test the program at the moment, so it would be great to know how you tested the program?
I haven't updated this project for a long time and I see a lot of changes in my `fixing_app` branch, although I wouldn't like to merge it into the main branch, since it looks very draft. So it would be great to know about tests, so that I can merge your changes and delete the `fixing_app` branch.
Thanks.

silphendio committed:
Use the stereo_pair function from `fixing_app`. It's faster.

silphendio commented:
I converted a few videos. The longest was over 30 minutes (where the old version ran out of memory).
I also tried to produce depth videos, and then make new 3d videos from that depth video. That worked too.
I'm not sure how to properly test this though. It's entirely possible that there are some bugs left.

The resulting video is always saved as mp4, audio is converted to aac.

The final video is just left&right images concenated together. All VR video players can play that. I changed the sides, so the left image is actially left, but even the cross-eyed view is supported by most players.

I haven't really tried the `fixing_app` branch, but I looked at some of the code.
It has a faster Stereo-conversion. I wasn't sure if I should commit that, but I did it just now in case you delete that branch.
